lms/djangoapps/teams/views.py

```diff
@@ -0,0 +1,448 @@
+    lookup_field = 'team_id'
+    serializer_class = CourseTeamSerializer
+    parser_classes = (MergePatchParser,)
+
+    def get_queryset(self):
+        """
+        Returns the queryset used to access the given team.
+        """
+        return CourseTeam.objects.all()
+
+
+class TopicListView(GenericAPIView):
+    """
+        **Use Cases**
+
+            Retrieve a list of topics associated with a single course.
+
+        **Example Requests**
+
+            GET /api/team/v0/topics/?course_id={course_id}
+
+        **Response Values for GET**
+
+            The following options can be specified as query parameters:
+
+            * course_id: Filters the result to topics belonging to the given course (required).
+
+            * order_by: Orders the results. Currently only 'name' is supported, and is also the default value.
+
+            * page_size: Number of results to return per page.
+
+            * page: Page number to retrieve.
+
+            If the course_id is not given or an unsupported value is passed for order_by, returns a 400 error.
+
+            If the user is not logged in or not enrolled in the course, returns a 403 error.
+
+            If the course does not exist, returns a 404 error.
+
+            Otherwise, a 200 response is returned containing the following fields:
+
+            * count: The total number of topics matching the request.
+
+            * next: The URL to the next page of results, or null if this is the last page.
+
+            * previous: The URL to the previous page of results, or null if this is the first page.
+
+            * num_pages: The total number of pages in the result.
+
+            * results: A list of the topics matching the request.
+
+                * id: The topic's unique identifier.
+
+                * name: The name of the topic.
+
+                * description: A description of the topic.
+
+    """
+
+    authentication_classes = (SessionAuthentication, OAuth2Authentication)
+    permission_classes = (permissions.IsAuthenticated,)
+
+    paginate_by = 10
+    paginate_by_param = 'page_size'
+    pagination_serializer_class = PaginationSerializer
+    serializer_class = TopicSerializer
+
+    def get(self, request):
+        """
+        GET /api/team/v0/topics/?course_id={course_id}
+        """
+        course_id_string = request.QUERY_PARAMS.get('course_id', None)
+        if course_id_string is None:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        try:
+            course_id = CourseKey.from_string(course_id_string)
+        except InvalidKeyError:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        course_module = modulestore().get_course(course_id)
+        if course_module is None:  # course is None if not found
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        if not CourseEnrollment.is_enrolled(request.user, course_id):
```

Comment:
@benmcmorran I just did a quick test by creating a new course, then hitting that endpoint as staff and verifying that it didn't 403. Either way, it's better to leave it explicitly checking for `is_staff`.